openedx/core/djangoapps/credit/models.py

```diff
@@ -201,13 +221,33 @@ class CreditRequirementStatus(TimeStampedModel):
     class Meta(object):  # pylint: disable=missing-docstring
         get_latest_by = "created"
 
+    @classmethod
+    def add_or_update_requirement_status(cls, username, requirement, status="satisfied", reason=None):
```

Comment:
We should probably wrap this in a transaction to ensure that the requirement is created AND the status is updated.